### PR TITLE
FIREFLY-1464: Cleanup Table Info dialog, metadata

### PR DESCRIPTION
**Pull request by jaladh-singhal**
Fixes [FIREFLY-1464](https://jira.ipac.caltech.edu/browse/FIREFLY-1464)

- Made font sizes same, aligned their baselines (note: everything is sm now, key value pairs have different font-weights to disambiguate)
- Replace "show more" link by chip "show"
- Made metadata info popup resizable
- Removed css styles and cleaned up redundant code & layout a bit

## Testing
https://fireflydev.ipac.caltech.edu/firefly-1464-metadata/firefly/

Go to TAP -> m81, 100 arcsec -> table info.
- compare font sizes and alignment with nightly build in both Job info and metadata tab 
- click on "show" chip button in "table metadata" tab
   - the popup should be resizable 
   - when popup size is reduced below content dimensions, the content should be scrollable
- click on "show" button in "job info" tab, the popup should be as before with minor improvements (to test css replaced by sx)

### CODE REVIEW

File: src/firefly/js/core/background/JobInfo.jsx [hunk 1]
```diff
@@ -83,8 +80,8 @@ function DataOrigin({dataOrigin, type, jobId}) {
     if (!isUws) return <KeywordBlock label={label} value={dataOrigin} asLink={true}/>;
     return (
         <div style={{display: 'inline-flex'}}>
-            <KeywordBlock style={{width: 425, alignItems:'center'}} label={label} value={dataOrigin} asLink={true}/>
-            <Button ml={1/2} onClick={() => showUwsJob({jobId})}>Show</Button>
```
Comment (jaladh-singhal) [inline src/firefly/js/core/background/JobInfo.jsx:87]: Should I also convert this to Chip for consistency with Table metadata tab?

**Comment by gpdf** [comment]
In general it looks so much better now, and has a clearer impact on the user!

The baseline of the text in the chip is still not aligned with the text preceding it.  I happen to be unusually sensitive to such things; I don't think it's essential to fix if it's difficult for some reason.  Still it seems like the sort of thing we could work out once and then apply everywhere.
 
image (10)

**Comment by gpdf** [comment]
There's something I don't understand about the construction of the tree.

The following (collapsed) tree

Screenshot 2024-04-23 at 7 37 15 PM

results from uploading the file appended below.  I don't understand why the data appear in the order they do -- the "type" and "utype" attributes of the <RESOURCE> element separated by the "groups", for instance.  I realize this is an issue we'll have to defer to [FIREFLY-1466](https://jira.ipac.caltech.edu/browse/FIREFLY-1466), but it's odd and it makes it harder to understand what one is seeing.  I'll add appropriate comments on that ticket, though.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<VOTABLE xmlns="http://www.ivoa.net/xml/VOTable/v1.3" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.4">
  <RESOURCE type="results">
    <INFO name="QUERY_STATUS" value="OK" />
    <INFO name="QUERY_TIMESTAMP" value="2024-04-16T19:56:50.119" />
    <INFO name="QUERY" value="SELECT coord_dec,coord_ra,objectId  FROM dp02_dc2_catalogs.Object  WHERE CONTAINS(POINT('ICRS', coord_ra, coord_dec),CIRCLE('ICRS', 62, -37, 0.06944444444444445))=1" />
    <TABLE>
      <FIELD name="coord_dec" datatype="double" ID="col_0" ucd="pos.eq.dec;meta.main" unit="deg">
        <DESCRIPTION>Fiducial ICRS Declination of centroid used for database indexing</DESCRIPTION>
      </FIELD>
      <FIELD name="coord_ra" datatype="double" ID="col_1" ucd="pos.eq.ra;meta.main" unit="deg">
        <DESCRIPTION>Fiducial ICRS Right Ascension of centroid used for database indexing</DESCRIPTION>
      </FIELD>
      <FIELD name="objectId" datatype="long" ID="col_2" ucd="meta.id;src;meta.main">
        <DESCRIPTION>Unique id. Unique ObjectID</DESCRIPTION>
      </FIELD>
      <DATA>
        <TABLEDATA>
          <TR>
            <TD>-36.9641438</TD>
            <TD>62.0727779</TD>
            <TD>1651281746966117391</TD>
          </TR>
          <TR>
            <TD>-36.9656694</TD>
            <TD>62.0369809</TD>
            <TD>1651281746966117378</TD>
          </TR>
          <TR>
            <TD>-36.965212</TD>
            <TD>62.036374</TD>
            <TD>1651281746966117379</TD>
          </TR>
          <TR>
            <TD>-36.9657936</TD>
            <TD>62.0738129</TD>
            <TD>1651281746966117386</TD>
          </TR>
          <TR>
            <TD>-36.9638082</TD>
            <TD>62.0729117</TD>
            <TD>1651281746966117388</TD>
          </TR>
          <TR>
            <TD>-36.9665291</TD>
            <TD>62.0709283</TD>
            <TD>1651281746966117390</TD>
          </TR>
          <TR>
            <TD>-36.9657835</TD>
            <TD>62.0365534</TD>
            <TD>1651281746966117377</TD>
          </TR>
          <TR>
            <TD>-36.9667397</TD>
            <TD>62.0723198</TD>
            <TD>1651281746966117393</TD>
          </TR>
          <TR>
            <TD>-36.9663113</TD>
            <TD>62.0726724</TD>
            <TD>1651281746966117394</TD>
          </TR>
          <TR>
            <TD>-36.9657731</TD>
            <TD>62.0713128</TD>
            <TD>1651281746966117396</TD>
          </TR>
          <TR>
            <TD>-36.9670318</TD>
            <TD>62.0751776</TD>
            <TD>1651281746966117397</TD>
          </TR>
          <TR>
            <TD>-36.9674324</TD>
            <TD>62.0757187</TD>
            <TD>1651281746966117399</TD>
          </TR>
          <TR>
            <TD>-36.963574</TD>
            <TD>62.0721459</TD>
            <TD>1651281746966117400</TD>
          </TR>
          <TR>
            <TD>-36.9663653</TD>
            <TD>62.075253</TD>
            <TD>1651281746966117402</TD>
          </TR>
          <TR>
            <TD>-36.9669228</TD>
            <TD>62.0713272</TD>
            <TD>1651281746966117403</TD>
          </TR>
          <TR>
            <TD>-36.9653659</TD>
            <TD>62.0749717</TD>
            <TD>1651281746966117405</TD>
          </TR>
          <TR>
            <TD>-36.9658675</TD>
            <TD>62.0042675</TD>
            <TD>1651281746966117372</TD>
          </TR>
          <TR>
            <TD>-36.9653025</TD>
            <TD>62.0035388</TD>
            <TD>1651281746966117373</TD>
          </TR>
          <TR>
            <TD>-36.9645116</TD>
            <TD>62.0363382</TD>
            <TD>1651281746966117376</TD>
          </TR>
          <TR>
            <TD>-36.9658251</TD>
            <TD>62.013897</TD>
            <TD>1651281746966117438</TD>
          </TR>
          <TR>
            <TD>-36.964871</TD>
            <TD>62.0249569</TD>
            <TD>1651281746966117430</TD>
          </TR>
          <TR>
            <TD>-36.9645238</TD>
            <TD>62.0243696</TD>
            <TD>1651281746966117431</TD>
          </TR>
          <TR>
            <TD>-36.9651199</TD>
            <TD>62.0239367</TD>
            <TD>1651281746966117432</TD>
          </TR>
          <TR>
            <TD>-36.9648839</TD>
            <TD>62.0052811</TD>
            <TD>1651281746966117433</TD>
          </TR>
          <TR>
            <TD>-36.9644356</TD>
            <TD>62.0041846</TD>
            <TD>1651281746966117434</TD>
          </TR>
          <TR>
            <TD>-36.966106</TD>
            <TD>62.0152371</TD>
            <TD>1651281746966117436</TD>
          </TR>
          <TR>
            <TD>-36.9647233</TD>
            <TD>62.0142316</TD>
            <TD>1651281746966117437</TD>
          </TR>
          <TR>
            <TD>-36.9649166</TD>
            <TD>62.0133083</TD>
            <TD>1651281746966117439</TD>
          </TR>
          <TR>
            <TD>-36.9641754</TD>
            <TD>62.0120927</TD>
            <TD>1651281746966117440</TD>
          </TR>
          <TR>
            <TD>-36.9669831</TD>
            <TD>62.0132374</TD>
            <TD>1651281746966117441</TD>
          </TR>
          <TR>
            <TD>-36.966804</TD>
            <TD>62.0135091</TD>
            <TD>1651281746966117442</TD>
          </TR>
          <TR>
            <TD>-36.9648614</TD>
            <TD>62.0148426</TD>
            <TD>1651281746966117443</TD>
          </TR>
          <TR>
            <TD>-36.9655012</TD>
            <TD>62.056352</TD>
            <TD>1651281746966117406</TD>
          </TR>
          <TR>
            <TD>-36.9654204</TD>
            <TD>62.0576667</TD>
            <TD>1651281746966117407</TD>
          </TR>
          <TR>
            <TD>-36.9657126</TD>
            <TD>62.0549251</TD>
            <TD>1651281746966117408</TD>
          </TR>
          <TR>
            <TD>-36.9657168</TD>
            <TD>62.0539208</TD>
            <TD>1651281746966117409</TD>
          </TR>
          <TR>
            <TD>-36.9651432</TD>
            <TD>62.0614615</TD>
            <TD>1651281746966117413</TD>
          </TR>
          <TR>
            <TD>-36.9655713</TD>
            <TD>62.0629522</TD>
            <TD>1651281746966117414</TD>
          </TR>
          <TR>
            <TD>-36.9651938</TD>
            <TD>62.0091366</TD>
            <TD>1651281746966117420</TD>
          </TR>
          <TR>
            <TD>-36.965739</TD>
            <TD>62.0499904</TD>
            <TD>1651281746966117297</TD>
          </TR>
          <TR>
            <TD>-36.9666456</TD>
            <TD>62.0562113</TD>
            <TD>1651281746966117284</TD>
          </TR>
          <TR>
            <TD>-36.9596563</TD>
            <TD>62.0518493</TD>
            <TD>1651281746966117285</TD>
          </TR>
          <TR>
            <TD>-36.9552907</TD>
            <TD>62.0553009</TD>
            <TD>1651281746966117286</TD>
          </TR>
          <TR>
            <TD>-36.9536951</TD>
            <TD>62.0488747</TD>
            <TD>1651281746966117287</TD>
          </TR>
          <TR>
            <TD>-36.9578527</TD>
            <TD>62.0511473</TD>
            <TD>1651281746966117288</TD>
          </TR>
          <TR>
            <TD>-36.9628627</TD>
            <TD>62.0497758</TD>
            <TD>1651281746966117289</TD>
          </TR>
          <TR>
            <TD>-36.9640702</TD>
            <TD>62.0479157</TD>
            <TD>1651281746966117290</TD>
          </TR>
          <TR>
            <TD>-36.9552354</TD>
            <TD>62.0517798</TD>
            <TD>1651281746966117291</TD>
          </TR>
          <TR>
            <TD>-36.9634927</TD>
            <TD>62.0495159</TD>
            <TD>1651281746966117292</TD>
          </TR>
          <TR>
            <TD>-36.9660329</TD>
            <TD>62.0507078</TD>
            <TD>1651281746966117293</TD>
          </TR>
        </TABLEDATA>
      </DATA>
    </TABLE>
    <INFO name="QUERY_STATUS" value="OVERFLOW" />
  </RESOURCE>
  <RESOURCE type="meta" utype="adhoc:service">
    <DESCRIPTION>Retrieve ForcedSource time series</DESCRIPTION>
    <PARAM name="accessURL" datatype="char" arraysize="*" value="https://data.lsst.cloud/api/datalink/timeseries" />
    <PARAM name="standardID" datatype="char" arraysize="*" value="lsst://api.data.lsst.cloud/datalink/timeseries#v0.1" />
    <GROUP name="inputParams">
      <PARAM name="id" datatype="long" ucd="meta.id" ref="col_2" value="">
        <DESCRIPTION>Object ID for time series</DESCRIPTION>
      </PARAM>
      <PARAM name="table" datatype="char" arraysize="*" value="dp02_dc2_catalogs.ForcedSource">
        <DESCRIPTION>Table containing time series data</DESCRIPTION>
      </PARAM>
      <PARAM name="id_column" datatype="char" arraysize="*" value="objectId">
        <DESCRIPTION>Foreign key in time series table</DESCRIPTION>
      </PARAM>
      <PARAM name="join_time_column" datatype="char" arraysize="*" value="dp02_dc2_catalogs.CcdVisit.expMidptMJD">
        <DESCRIPTION>Foreign column spec for time retrieval</DESCRIPTION>
      </PARAM>
      <PARAM name="band" datatype="char" arraysize="*" value="all">
        <DESCRIPTION>Filter band to retrieve (default: all)</DESCRIPTION>
        <VALUES>
          <OPTION value="all" />
          <OPTION value="u" />
          <OPTION value="g" />
          <OPTION value="r" />
          <OPTION value="i" />
          <OPTION value="z" />
          <OPTION value="y" />
        </VALUES>
      </PARAM>
      <PARAM name="detail" datatype="char" arraysize="*" value="principal">
        <DESCRIPTION>Level of detail for time series (default: principal)</DESCRIPTION>
        <VALUES>
          <OPTION value="full" />
          <OPTION value="principal" />
          <OPTION value="minimal" />
        </VALUES>
      </PARAM>
    </GROUP>
  </RESOURCE>
</VOTABLE>
```

File: src/firefly/js/core/background/JobInfo.jsx (hunk 1 above)
Comment (gpdf) [reply src/firefly/js/core/background/JobInfo.jsx:87]: I think the uppercase "Show" fits better with the style of the Job Info tab, and the lowercase one fits better with the style of the Metadata tab, where in general everything is lowercase.  So I don't think it needs to be changed.

Incidentally, the text baselines are perfectly aligned in the Job Info version of this:

image

**Review by gpdf — APPROVED** [review]
Addresses the "high" and "medium" parts of FIREFLY-1464.

OK with deferring the "low" priority parts to FIREFLY-1466, especially as they are in a different area of the code.

One suggestion for a remaining improvement to baseline alignment remains in the specific comments above.  Please fix if easy; if there's something about JoyUI that makes it awkward or it's otherwise messy, you can ignore.

**Comment by jaladh-singhal** [comment]
> In general it looks so much better now, and has a clearer impact on the user!
> 
> The baseline of the text in the chip is still not aligned with the text preceding it. I happen to be unusually sensitive to such things; I don't think it's essential to fix if it's difficult for some reason. Still it seems like the sort of thing we could work out once and then apply everywhere.
> 
> image (10)

@gpdf good catch - fixed it

File: src/firefly/js/tables/ui/ObjectTree.jsx [hunk 2]
```diff
@@ -30,10 +32,9 @@ ObjectTree.propTypes = {
     defaultExpandAll:   PropTypes.bool,
     onSelect:           PropTypes.func,
     selectable:         PropTypes.bool,
-    className:          PropTypes.string,
     nodeClassName:      PropTypes.string,
-    style:              PropTypes.object,
-    title:              PropTypes.node
+    sx:                 PropTypes.object,
+    title:              PropTypes.oneOfType([PropTypes.node, PropTypes.string])
```
Comment (loitly) [inline src/firefly/js/tables/ui/ObjectTree.jsx:37]: Why specifically define `string` or `node` when node already includes string?

**Review by loitly — APPROVED** [review]
It looks good. However, I noticed that the `Resources` section appeared with a `show` button even when there are no resources. Is this right?

image

**Comment by jaladh-singhal** [comment]
> I don't understand why the data appear in the order they do -- the "type" and "utype" attributes of the element separated by the "groups", for instance.

@gpdf I think it has to do with how firefly represents this as object internally (which is not order sensitive) - but @loitly knows better.

File: src/firefly/js/tables/ui/ObjectTree.jsx (hunk 2 above)
Comment (jaladh-singhal) [reply src/firefly/js/tables/ui/ObjectTree.jsx:37]: Oh right. I think I confused the two because I'm explicitly converting string to Typography element

**Comment by jaladh-singhal** [comment]
> It looks good. However, I noticed that the `Resources` section appeared with a `show` button even when there are no resources. Is this right?
> 
> image

@loitly what query/upload you make? I need to reproduce this to check

**Comment by gpdf** [comment]
> > It looks good. However, I noticed that the `Resources` section appeared with a `show` button even when there are no resources. Is this right?
> > image
> 
> @loitly what query/upload you made? I need to reproduce this to check

Curious about this, too.

File: src/firefly/js/tables/ui/ObjectTree.jsx (hunk 2 above)
Comment (jaladh-singhal) [reply src/firefly/js/tables/ui/ObjectTree.jsx:37]: I'll change node to element (I confused the two again 😄)

**Pull request merged**